src/ai/backend/manager/idle.py

```diff
@@ -487,27 +466,32 @@ async def set_remaining_time_report(
         )
 
 
-class AbstractIdleChecker(metaclass=ABCMeta):
-    terminate_reason: KernelLifecycleEventReason
+class _AbstractIdleChecker(metaclass=ABCMeta):
```

Comment:
Why did you make the Abstract class private? And ABC inheritance seems more appropriate.

Reply:
I changed it because it is not used in any other modules

Reply:
From a scalability perspective, there seems to be no reason not to expose the abstract class as public.
Rather than focusing on whether it is currently used or not, please decide on its visibility (public or private) based on the design intent.